patches/v8/revert_api_dcheck-fail_when_we_reenter_v8_while_terminating.patch

```diff
@@ -0,0 +1,142 @@
+From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
+From: clavin <clavin@DESKTOP-QVNUS47.fastel.com>
+Date: Thu, 7 Dec 2023 16:11:40 -0500
+Subject: Revert "[api] DCHECK-fail when we reenter V8 while terminating"
+
+This reverts commit 3eed17c6a12a0f33639aa5cffb822ebe42931ec8.
+
+This change causes Electron to crash. :(
+
+This patch is meant to be temporary and should be removed after more
+investigation and potentially a refactor.
+
+diff --git a/src/api/api-macros.h b/src/api/api-macros.h
+index fa272148c4553a2cbfe2438b2770d2fe5cde9a38..08fa9f799b217eb1565d28aa4f86eb5e0001422e 100644
+--- a/src/api/api-macros.h
++++ b/src/api/api-macros.h
+@@ -48,7 +48,9 @@
+ #define ENTER_V8_HELPER_INTERNAL(i_isolate, context, class_name,    \
+                                  function_name, bailout_value,      \
+                                  HandleScopeClass, do_callback)     \
+-  DCHECK(!i_isolate->is_execution_terminating());                   \
++  if (i_isolate->is_execution_terminating()) {                      \
++    return bailout_value;                                           \
++  }                                                                 \
+   HandleScopeClass handle_scope(i_isolate);                         \
+   CallDepthScope<do_callback> call_depth_scope(i_isolate, context); \
+   API_RCS_SCOPE(i_isolate, class_name, function_name);              \
+@@ -56,7 +58,9 @@
+   bool has_pending_exception = false
+ 
+ #define PREPARE_FOR_DEBUG_INTERFACE_EXECUTION_WITH_ISOLATE(i_isolate, T)       \
+-  DCHECK(!i_isolate->is_execution_terminating());                              \
++  if (i_isolate->is_execution_terminating()) {                                 \
++    return MaybeLocal<T>();                                                    \
++  }                                                                            \
+   InternalEscapableScope handle_scope(i_isolate);                              \
+   CallDepthScope<false> call_depth_scope(i_isolate, v8::Local<v8::Context>()); \
+   i::VMState<v8::OTHER> __state__((i_isolate));                                \
+diff --git a/test/unittests/execution/thread-termination-unittest.cc b/test/unittests/execution/thread-termination-unittest.cc
+index cce3969a127355643f2d872f45adbe5007062a2c..4c7c23c304f4d5b3c1833de73ca04ccf21892077 100644
+--- a/test/unittests/execution/thread-termination-unittest.cc
++++ b/test/unittests/execution/thread-termination-unittest.cc
+@@ -372,6 +372,99 @@ TEST_F(ThreadTerminationTest, TerminateLoadICException) {
+ Persistent<String> reenter_script_1;
+ Persistent<String> reenter_script_2;
+ 
++void ReenterAfterTermination(const FunctionCallbackInfo<Value>& info) {
```

Comment:
also nit: given we don't run these tests we can just revert the behavior change for a smaller patch

Reply:
Good catch. Ensmallified in 41416fe4b0b76e6dbf2de0bb113af4d9730b851f